src/sumcheck/mod.rs

```diff
@@ -1 +1,405 @@
+                .elementwise_sum(&combined_quad.bind(&bindings[1]).scale(alpha));
+
+            // Specification interpretation verification: Because the length of g is the same as the
+            // number of bits needed to describe wires on this layer (logw), bound_quad[g, l, r] = 0
+            // for any g > 0. Thus bound_quad is effectively two-dimensional.
+            for index in 1..bound_quad.len() {
+                assert!(bound_quad[index].is_empty());
+            }
+
+            // Allocate room for the new bindings this layer will generate
+            let mut new_bindings = [
+                vec![FE::ZERO; layer.logw.into()],
+                vec![FE::ZERO; layer.logw.into()],
+            ];
+
+            // Generate the pad for this layer. The pad has the same structure as the proof since the
+            // one has to be substracted from the other.
+            let layer_pad = ProofLayer {
+                polynomials: vec![
+                    [Polynomial {
+                        p0: pad_generator(),
+                        p2: pad_generator(),
+                    }; 2];
+                    layer.logw.into()
+                ],
+                vl: pad_generator(),
+                vr: pad_generator(),
+            };
+
+            // Allocate the proof for this layer. The zero values in the polynomial are not
+            // significant. We just need an initial value.
+            let mut layer_proof_polynomials = vec![
+                [Polynomial {
+                    p0: FE::ZERO,
+                    p2: FE::ZERO
+                }; 2];
+                layer.logw.into()
+            ];
+
+            // (VL, VR) = wires
+            // The specification says "wires[j]" where 0 <= j < circuit.num_layers. Recall that
+            // evaluation.wires includes the circuit output wires at index 0 so we have to go up one
+            // to get the input wires for this layer.
+            // This makes sense because over the course of the loop that follows, we'll bind each of
+            // left_ and right_wires to a challenge layer.logw times, exactly enough to reduce these
+            // arrays to a single element, which become the layer commitments vl and vr.
+            let mut left_wires = evaluation.wires[layer_index + 1].clone();
+            let mut right_wires = evaluation.wires[layer_index + 1].clone();
+
+            for (round, (pad_polynomials, proof_polynomials)) in layer_pad
+                .polynomials
+                .into_iter()
+                .zip(&mut layer_proof_polynomials)
+                .enumerate()
+            {
+                for hand in 0..2 {
+                    // Implements the polynomial from the specification:
+                    // Let p(x) = SUM_{l, r} bind(QUAD, x)[l, r] * bind(VL, x)[l] * VR[r]
+                    let evaluate_polynomial = |at: FE| {
+                        let bind = &[at];
+                        let bound_quad_at = bound_quad.bind(bind);
```

Comment:
Per our discussion, I think we might have to change this to bind QUAD along the left hand dimension. Currently this is just doing a scalar multiplication.
```suggestion
                        let bound_quad_at = bound_quad[0].bind(bind);
```